cal/views.py

```diff
@@ -53,15 +53,24 @@ def formatday(self, day, weekday):
             return self.day_cell(cssclass, day)
         return self.day_cell('noday', '&nbsp;')
 
-    def formatmonth(self, year, month):
-        self.year, self.month = year, month
+    def formatmonthname(self, theyear, themonth, withyear=True):
```

Comment:
hey @thp thanks for the commit, while changing the function name, you forgot do adapt the currentmonth function with the new function call https://github.com/Metalab/mos/blob/c7fed603467ace6c46a2993c1833a9fe7ed1dc9f/cal/views.py#L86 could you adapt this one? would be awesome and looking forward to merge the fixed patch! :)

Reply:
The diff might indeed be a bit misleading, so let me elaborate:

1. I removed the `formatmonth()` override so that the [original function](https://github.com/python/cpython/blob/e57ecf6bbc59f999d27b125ea51b042c24a07bd9/Lib/calendar.py#L510) in the library is used (which in turn calls `formatmonthname()`)
2. Instead, I did an override of `formatmonthname()` that was previously this [original function](https://github.com/python/cpython/blob/e57ecf6bbc59f999d27b125ea51b042c24a07bd9/Lib/calendar.py#L499)

In other words, the current code does the following:

1. `self.formatmonth(...)` is called
2. `.formatmonth()` has a subclass override, which adds the navigation links to the top (_this part is removed_)
3. It then calls the base class `.formatmonth()` to do the table calendar
4. The base class `.formatmonth()` is called, and in turn calls `.formatmonthname()`

The proposed PR changes this to:

1. `self.formatmonth(...)` is called
2. The base class `.formatmonth()` is called, and in turn calls `.formatmonthname()`
3. The `.formatmonthname()` function now has a subclass override and adds the nav links (_this part is added_)

The diff looks as if the function was renamed, because the old override (of `.formatmonth()`) and the new override (of `.formatmonthname()`) share two lines of code:

```
        prev = d - relativedelta.relativedelta(months=1)
        next = d + relativedelta.relativedelta(months=1)
```

The base class which gets its member functions overridden is [`calendar.HTMLCalendar`](https://docs.python.org/3/library/calendar.html#calendar.HTMLCalendar), see [`Lib/calendar.py`](https://github.com/python/cpython/blob/main/Lib/calendar.py) in CPython.

With that said, I still haven't tested it locally, so if you could test it in a dev environment it before merging and deploying that'd be great.

Reply:
thank you for your awesome elaboration! :)

I will test it locally today and merge it if its works!

Thanks a lot @thp !